Exception on akka.dispatch.NonBlockingBoundedMailbox

When configuring actor to use mailbox with following configuration:
```
bounded-node-queue {
  mailbox-type = "akka.dispatch.NonBlockingBoundedMailbox"
  mailbox-capacity = 5
}
```

and running an example application we get an error:
```
[info] 2022-04-11 21:23:15 ERROR a.a.t.internal.adapter.ActorAdapter akka://mailbox-simple/user/receive-bounded-node-queue swallowing exception during message send
[info] java.lang.IllegalStateException: VirtualField not found
[info] 	at io.opentelemetry.javaagent.tooling.field.RuntimeFieldBasedImplementationSupplier.find(RuntimeFieldBasedImplementationSupplier.java:27)
[info] 	at io.opentelemetry.javaagent.shaded.instrumentation.api.field.VirtualField.find(VirtualField.java:40)
[info] 	at io.scalac.mesmer.otelextension.instrumentations.akka.actor.AbstractBoundedNodeQueueAdvice$.add(AkkaMailboxInstrumentations.scala:62)
[info] 	at akka.dispatch.AbstractBoundedNodeQueue.add(AbstractBoundedNodeQueue.java:110)
[info] 	at akka.dispatch.BoundedNodeMessageQueue.enqueue(Mailbox.scala:426)
[info] 	at akka.dispatch.Mailbox.enqueue(Mailbox.scala:89)
[info] 	at akka.dispatch.Dispatcher.dispatch(Dispatcher.scala:63)
[info] 	at akka.actor.dungeon.Dispatch.sendMessage(Dispatch.scala:163)
[info] 	at akka.actor.dungeon.Dispatch.sendMessage$(Dispatch.scala:157)
[info] 	at akka.actor.ActorCell.sendMessage(ActorCell.scala:411)
[info] 	at akka.actor.Cell.sendMessage(ActorCell.scala:327)
[info] 	at akka.actor.Cell.sendMessage$(ActorCell.scala:326)
[info] 	at akka.actor.ActorCell.sendMessage(ActorCell.scala:411)
[info] 	at akka.actor.LocalActorRef.$bang(ActorRef.scala:448)
[info] 	at akka.actor.typed.internal.adapter.ActorRefAdapter.tell(ActorRefAdapter.scala:27)
[info] 	at akka.actor.typed.ActorRef$ActorRefOps$.$bang$extension(ActorRef.scala:67)
[info] 	at example.GeneratorBehavior$.$anonfun$apply$5(SimpleMailboxExample.scala:73)
[info] 	at example.GeneratorBehavior$.$anonfun$apply$5$adapted(SimpleMailboxExample.scala:73)
[info] 	at scala.collection.immutable.List.foreach(List.scala:333)
[info] 	at example.GeneratorBehavior$.$anonfun$apply$4(SimpleMailboxExample.scala:73)
[info] 	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:190)
[info] 	at example.GeneratorBehavior$.$anonfun$apply$3(SimpleMailboxExample.scala:73)
[info] 	at akka.actor.typed.internal.BehaviorImpl$ReceiveMessageBehavior.receive(BehaviorImpl.scala:153)
[info] 	at akka.actor.typed.Behavior$.interpret(Behavior.scala:274)
[info] 	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:230)
[info] 	at akka.actor.typed.internal.adapter.ActorAdapter.handleMessage(ActorAdapter.scala:124)
[info] 	at akka.actor.typed.internal.adapter.ActorAdapter.aroundReceive(ActorAdapter.scala:107)
[info] 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:580)
[info] 	at akka.actor.ActorCell.invoke(ActorCell.scala:548)
[info] 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
[info] 	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
[info] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info] 	at java.base/java.lang.Thread.run(Thread.java:829)
[info] Caused by: java.lang.ClassNotFoundException: io/opentelemetry/javaagent/bootstrap/field/VirtualFieldImpl$akka$dispatch$AbstractBoundedNodeQueue$boolean
[info] 	at java.base/java.lang.Class.forName0(Native Method)
[info] 	at java.base/java.lang.Class.forName(Class.java:398)
[info] 	at io.opentelemetry.javaagent.tooling.field.RuntimeFieldBasedImplementationSupplier.find(RuntimeFieldBasedImplementationSupplier.java:23)
[info] 	... 33 common frames omitted
```

--- a/otel-extension/src/main/scala/io/scalac/mesmer/otelextension/instrumentations/akka/actor/AkkaMailboxInstrumentations.scala
+++ b/otel-extension/src/main/scala/io/scalac/mesmer/otelextension/instrumentations/akka/actor/AkkaMailboxInstrumentations.scala
@@ -1,5 +1,7 @@
 package io.scalac.mesmer.otelextension.instrumentations.akka.actor
 
+import java.lang.{ Boolean => JBoolean }
+
 import akka.MesmerMirrorTypes.ActorRefWithCell
 import akka.MesmerMirrorTypes.Cell
 import akka.actor.BoundedQueueProxy
@@ -31,7 +33,7 @@ object BoundedNodeMessageQueueAdvice {
       case _: BoundedNodeMessageQueue =>
         incDropped(
           VirtualField
-            .find(classOf[AbstractBoundedNodeQueue[_]], classOf[Boolean])
+            .find(classOf[AbstractBoundedNodeQueue[_]], classOf[JBoolean])
             .get(self.asInstanceOf[AbstractBoundedNodeQueue[_]]),
           withCell.underlying
         )
@@ -45,11 +47,10 @@ object BoundedNodeMessageQueueAdvice {
 
   @inline
   private def incDropped(result: Boolean, cell: Cell): Unit =
-    if (result && (cell ne null)) {
+    if (!result && (cell ne null)) {
       val maybeActorMetrics = ActorCellDecorator.getMetrics(cell)
       for {
         actorMetrics <- maybeActorMetrics if actorMetrics.droppedMessages.isDefined
-
       } actorMetrics.droppedMessages.get.inc()
     }
 }
@@ -59,7 +60,7 @@ object AbstractBoundedNodeQueueAdvice {
   @OnMethodExit
   def add(@Return result: Boolean, @Advice.This self: Object): Unit =
     VirtualField
-      .find(classOf[AbstractBoundedNodeQueue[_]], classOf[Boolean])
+      .find(classOf[AbstractBoundedNodeQueue[_]], classOf[JBoolean])
       .set(self.asInstanceOf[AbstractBoundedNodeQueue[_]], result)
 
 }

--- a/otel-extension/src/test/scala/io/scalac/mesmer/instrumentation/akka/actor/ActorMailboxTest.scala
+++ b/otel-extension/src/test/scala/io/scalac/mesmer/instrumentation/akka/actor/ActorMailboxTest.scala
@@ -83,25 +83,25 @@ class ActorMailboxTest
           |
           |bounded-node-queue {
           |  mailbox-type = "akka.dispatch.NonBlockingBoundedMailbox"
-          |  mailbox-capacity = 5
+          |  mailbox-capacity = 3
           |}
           |
           |bounded-queue {
           |  mailbox-type = "akka.dispatch.BoundedMailbox"
           |   mailbox-push-timeout-time=0
-          |  mailbox-capacity = 5
+          |  mailbox-capacity = 3
           |}
           |
           |bounded-priority-queue {
           |  mailbox-type = "io.scalac.mesmer.instrumentation.akka.actor.HashCodePriorityMailbox"
           |  mailbox-push-timeout-time=1
-          |  mailbox-capacity = 5
+          |  mailbox-capacity = 3
           |}
           |
           |bounded-stable-priority-queue {
           |  mailbox-type = "io.scalac.mesmer.instrumentation.akka.actor.StableHashCodePriorityMailbox"
           |  mailbox-push-timeout-time=1
-          |  mailbox-capacity = 5
+          |  mailbox-capacity = 3
           |}
           |""".stripMargin
       )
@@ -135,7 +135,7 @@ class ActorMailboxTest
     (sut, context)
   }
 
-  private def testWithProps(props: Props, amountOfMessages: Int = 10, expectedValue: Long = 5): Any = {
+  private def testWithProps(props: Props, amountOfMessages: Int = 10, expectedValue: Long = 7): Any = {
 
     val (sut, context) = actorRefWithContext(props)
 

